core/shared/src/main/scala/org/http4s/Message.scala

```diff
@@ -246,18 +252,77 @@ sealed trait Message[F[_]] extends Media[F] { self =>
   override def covary[F2[x] >: F[x]]: SelfF[F2] = this.asInstanceOf[SelfF[F2]]
 
   /** Compiles the body stream to a single chunk and sets it as the
-    * body.  Replaces any `Transfer-Encoding: chunked` with a
-    * `Content-Length` header.  It is the caller's responsibility to
-    * assure there is enough memory to materialize the body.
+    * body. Replaces any `Transfer-Encoding: chunked` with a
+    * `Content-Length` header. It is the caller's responsibility to
+    * assure there is enough memory to materialize the body. Also,
+    * potentially this method could run infinitely.
+    * Consider to use [[org.http4s.Message.toStrict]] instead.
     */
-  def toStrict(implicit F: Concurrent[F]): F[Self] =
+  def unsafeToStrict(implicit F: Concurrent[F]): F[Self] =
     body.compile.to(Chunk).map { chunk =>
       withBodyStream(Stream.chunk(chunk))
         .transformHeaders(_.withContentLength(`Content-Length`.unsafeFromLong(chunk.size.toLong)))
     }
+
+  /** Compiles the body stream to a single chunk and sets it as the
+    * body. Replaces any `Transfer-Encoding: chunked` with a
+    * `Content-Length` header. It is the caller's responsibility to
+    * assure there is enough memory to materialize the entity body and
+    * set an appropriate timeout.
+    *
+    * @param timeout duration to wait for the entity stream completion.
+    *                Exceeding finite timeout duration
+    *                (an instance of [[FiniteDuration]]) will lead to
+    *                failing processing with the `TimeoutException`
+    * @param maxBytes maximum length of the entity stream. If the stream
+    *                 exceeds the limit then processing fails with the [[EntityStreamException]]
+    */
+  def toStrict(timeout: Duration, maxBytes: Option[Long])(implicit F: Temporal[F]): F[Self] = {
```

Comment:
Timeout is orthogonal: we can just apply it to the result of this.  I guess maxBytes is handy, but that too could be an orthogonal operation.  And then we're back to the original `toStrict`, which was fine by me.

Reply:
I agree that timeout is orthogonal, fewer about the size limiting (because it goes first when transforming the entity's body). But my whole point is that it's better to slightly force users to make a deliberate decision when using `toStrict`.

Reply:
I think I'd be happy with a `toStrict(maxBytes: Option[Long])`.  The alternative would be an `limitEntity(maxBytes: Long)` that composes with `.toStrict`.

Reply:
Touche 🤝  Let's make it so.